otx/algorithms/detection/tasks/inference.py

```diff
@@ -226,28 +229,33 @@ def _init_recipe(self):
         if train_type not in (TrainType.SEMISUPERVISED, TrainType.INCREMENTAL):
             raise NotImplementedError(f"Train type {train_type} is not implemented yet.")
         if train_type == TrainType.SEMISUPERVISED:
-            if self._data_cfg.get("data", None) and self._data_cfg.data.get("unlabeled", None):
-                recipe = os.path.join(recipe_root, "semisl.py")
+            if self._is_training:
+                if self._data_cfg.get("data", None) and self._data_cfg.data.get("unlabeled", None):
+                    recipe = os.path.join(recipe_root, "semisl.py")
+                    self.base_dir = os.path.join(self.template_dir, "semisl")
+                    self.data_pipeline_path = os.path.join(self.base_dir, "data_pipeline.py")
+                else:
+                    logger.warning("Cannot find unlabeled data.. convert to INCREMENTAL.")
```

Comment:
Shouldn't [SEMISUPERVISED + no unlabeled data] raise an error?

Reply:
In fact, there is no problem to see, but if the user misses that log, the user may still think that SEMI is running well.. It was just such a small worry.